### PR TITLE
Migrate IexecMaintenance unit tests

### PR DESCRIPTION
**Pull request by jeremyjams**

### CODE REVIEW

File: test/byContract/IexecMaintenance/IexecMaintenance.ts [hunk 1]
```diff
@@ -0,0 +1,186 @@
...
+const configureArgs = [
+    randomAddress(),
+    'some name',
+    'some symbol',
+    100,
+    randomAddress(),
+    randomAddress(),
+    randomAddress(),
+    randomAddress(),
+] as [string, string, string, number, string, string, string, string];
```
Comment (zguesmi) [inline test/byContract/IexecMaintenance/IexecMaintenance.ts:27]:
```suggestion
const configureParams = {
    token: randomAddress(),
    name: 'some name',
    symbol: 'some symbol',
    decimals: 100,
    appregistry: randomAddress(),
    datasetregistry: randomAddress(),
    workerpoolregistry: randomAddress(),
    m_v3_iexecHub: randomAddress(),
}
const configureArgs = Object.values(configureParams) as [string, string, string, number, string, string, string, string];
```
and then:
```
expect(await iexecPoco.token()).equal(configureParams.token);
```

File: test/byContract/IexecMaintenance/IexecMaintenance.ts [hunk 2]
```diff
@@ -0,0 +1,186 @@
...
+        iexecPocoAsAdmin = iexecPoco.connect(iexecAdmin);
+    }
+
+    describe('Configure', () => {
+        it('Should configure', async () => {
+            await setDomainSeparator(hashZero);
+            await iexecPocoAsAdmin.configure(...configureArgs).then((tx) => tx.wait());
+            expect(await iexecPoco.eip712domain_separator()).equal(
+                await hashDomain(await iexecPoco.domain()),
+            );
```
Comment (zguesmi) [inline test/byContract/IexecMaintenance/IexecMaintenance.ts:54]: Also compare with hash of local values.

File: test/byContract/IexecMaintenance/IexecMaintenance.ts [hunk 3]
```diff
@@ -0,0 +1,186 @@
...
+            expect(domain.chainId).equal((await ethers.provider.getNetwork()).chainId);
+            expect(domain.verifyingContract).equal(proxyAddress);
+        });
+    });
+
+    describe('Update domain separator', () => {
+        it('Should update domain separator', async () => {
+            await setDomainSeparator(someDomainSeparator);
+            expect(await iexecPoco.eip712domain_separator()).equal(someDomainSeparator);
+            await iexecPocoAsAdmin.updateDomainSeparator().then((tx) => tx.wait());
```
Comment (zguesmi) [inline test/byContract/IexecMaintenance/IexecMaintenance.ts:91]:
```suggestion
            await iexecPoco.updateDomainSeparator().then((tx) => tx.wait());
```
and others

File: test/byContract/IexecMaintenance/IexecMaintenance.ts [hunk 4]
```diff
@@ -0,0 +1,186 @@
...
+                'Ownable: caller is not the owner',
+            );
+        });
+    });
+
+    describe('Change registries', () => {
+        let iexecMaintenanceExtra: IexecMaintenanceExtra;
+
+        beforeEach('Deploy', async () => {
+            iexecMaintenanceExtra = IexecMaintenanceExtra__factory.connect(proxyAddress, anyone);
```
Comment (zguesmi) [inline test/byContract/IexecMaintenance/IexecMaintenance.ts:141]: Put this in `initFixture`

File: test/byContract/IexecMaintenance/IexecMaintenance.ts [hunk 5]
```diff
@@ -0,0 +1,186 @@
...
+                iexecMaintenanceExtra.changeRegistries(
+                    randomAddress(),
+                    randomAddress(),
+                    randomAddress(),
+                ),
+            ).to.be.revertedWith('Ownable: caller is not the owner');
+        });
+    });
+
+    async function setDomainSeparator(domainSeparator: string) {
```
Comment (zguesmi) [inline test/byContract/IexecMaintenance/IexecMaintenance.ts:168]:
```suggestion
    async function setDomainSeparatorInStorage(domainSeparator: string) {
```
or
```suggestion
    async function overrideDomainSeparatorInStorage(domainSeparator: string) {
```

File: test/byContract/IexecMaintenance/IexecMaintenance.ts (hunk 2 above)
Comment (jeremyjams) [reply test/byContract/IexecMaintenance/IexecMaintenance.ts:66]: Domain is already tested in `Should get domain`

**Review by zguesmi — APPROVED** [review]
Thanks for the updates.
I think you missed one comment (https://github.com/iExecBlockchainComputing/PoCo/pull/100#discussion_r1701988509).
Otherwise all good!

**Review by gfournieriExec — APPROVED** [review]
Thanks !

**Pull request merged**